src/disk.py

```diff
@@ -47,3 +48,28 @@ def write_event(event: dict):
         if not file_exists:
             writer.writeheader()
         writer.writerow(event)
+
+
+def cleanup_old_files():
+    """Delete CSV files older than 6 months."""
+    logger.info("Starting cleanup of old files")
+    cutoff = datetime.now() - timedelta(days=180)
+    deleted = 0
+
+    def scan_and_cleanup(path: pathlib.Path):
```

Comment:
Using a closure here is strange - why not just make this part of the parent function, or a separate function in and of itself?

Reply:
if you're going to use recursion, though, I think it would be better to have the function return the number of files deleted rather than modifying the `nonlocal` variable `deleted`. This would also allow you to re-enable the F824 warning which I think is generally useful